src/clr-angular/utils/a11y/aria-live.service.ts

```diff
@@ -0,0 +1,95 @@
+import { Inject, Injectable, NgZone, OnDestroy } from '@angular/core';
+import { uniqueIdFactory } from '../id-generator/id-generator.service';
+
+// export type AriaLivePoliteness = 'off' | 'polite' | 'assertive';
+export enum AriaLivePoliteness {
+  off = 'off',
+  polite = 'polite',
+  assertive = 'assertive',
+}
+
+@Injectable()
+export class AriaLiveService implements OnDestroy {
+  private _TICK: number = 100;
+  private _ariaLiveElement: HTMLElement;
+  private _document: Document;
+  private _previousTimeout?: number;
+
+  private _id: string = `clr-aria-live-element-${uniqueIdFactory()}`;
+
+  constructor(private _ngZone: NgZone, @Inject(DOCUMENT) _document: any) {
+    this._document = _document;
+    this._ariaLiveElement = this._createLiveElement();
+  }
+
+  private _createLiveElement(): HTMLElement {
+    const ariaLiveElement = this._document.createElement('div');
+
+    ariaLiveElement.setAttribute('id', this.getId());
+    // Use clarity screen reader class to hide the dom element
+    // and fix the scrollbar shake
+    ariaLiveElement.classList.add('clr-sr-only');
+
+    ariaLiveElement.setAttribute('aria-atomic', 'true');
+    ariaLiveElement.setAttribute('aria-live', AriaLivePoliteness.polite);
+
+    this._document.body.appendChild(ariaLiveElement);
+
+    return ariaLiveElement;
+  }
+
+  public announce(message: string, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): Promise<void> {
+    this.clear();
+    clearTimeout(this._previousTimeout);
+
+    this._ariaLiveElement.setAttribute('aria-live', politeness);
+
+    // This 100ms timeout is necessary for some browser + screen-reader combinations:
+    // - Both JAWS and NVDA over IE11 will not announce anything without a non-zero timeout.
+    // - With Chrome and IE11 with NVDA or JAWS, a repeated (identical) message won't be read a
+    //   second time without clearing and then using a non-zero delay.
+    // (using JAWS 17 at time of this writing).
+    return this._ngZone.runOutsideAngular(() => {
+      return new Promise(resolve => {
+        clearTimeout(this._previousTimeout);
+        /**
+         * Cause I didn't want to use `window.setTimeout` so TypeScript will stop throwing
+         * this error `Type 'Timer' is not assignable to type 'number'` , I had to use `<any>` before
+         * setTimeout()
+         *
+         * More on the topic here: https://github.com/TypeStrong/atom-typescript/issues/1053
+         */
+        this._previousTimeout = <any>setTimeout(() => {
+          this._ariaLiveElement.textContent = message;
+          resolve();
+        }, this._TICK);
+      });
+    });
+  }
+
+  public clear() {
+    if (this._ariaLiveElement) {
+      this._ariaLiveElement.textContent = '';
+    }
+  }
+
+  public ngOnDestroy() {
+    clearTimeout(this._previousTimeout);
+
+    if (this._ariaLiveElement && this._ariaLiveElement.parentNode) {
+      this._ariaLiveElement.parentNode.removeChild(this._ariaLiveElement);
```

Comment:
You appended the child to the body element(on line: 43) so we know it's in the body element. Just do: 
`this.document.removeChild(this._ariaLiveElement);`

Also, we need to check the platform here.

Reply:
Yep this could be optimized